aiida/cmdline/commands/cmd_daemon.py

```diff
@@ -214,7 +214,12 @@ def restart(ctx, reset, no_wait):
 
     if reset:
         ctx.invoke(stop)
-        ctx.invoke(start)
+        # These two lines can be simplified to `ctx.invoke(start)` once issue #950 in `click` is resolved.
+        # Due to that bug, the `callback` of the `number` argument the `start` command is not being called, which is
+        # responsible for settting the default value, which causes `None` to be passed and that triggers an exception.
+        # As a temporary workaround, we fetch the default here manually and pass that in explicitly.
+        number = ctx.obj.config.get_option('daemon.default_workers', ctx.obj.profile.name)
```

Comment:
Just to double check, `ctx.obj.profile.name` is always set even if not explicitly specified?

Reply:
Barring a bug, it should be. It is set in `aiida.cmdline.commands.cmd_verdi.verdi`